front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -97,34 +98,70 @@ const DEFAULT_PAYLOAD: Pick<
   start_time: '2024-07-15T08:00:00+02:00',
 };
 
-const createPathItemFromNode = (node: Node, index: number) => {
+const createPathItemFromNode = async (
+  node: Node,
+  index: number,
+  infraId: number,
+  dispatch: AppDispatch
+) => {
   const [trigram, secondaryCode] = node.betriebspunktName.split('/');
-  return {
+  let finalSecondaryCode: string | undefined;
+  if (!secondaryCode) {
```

Comment:
Thanks for the PR ! One suggestion: you could use an **early return** when secondaryCode is present. While it’s not mandatory this will reduce the nesting of if statements and make the code cleaner by simplifying the logic before the final return.
```suggestion
    if (secondaryCode) {
      return {
          trigram: trigram,               
          secondary_code: secondaryCode,
          id: `${node.id}-${index}`,    
      };
  }
// continue If no secondaryCode, prepare the search payload
```

Reply:
Thanks for the suggestions !
Its done.